test/person_memberships_test.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+# # frozen_string_literal: true
+require_relative './test_helper'
+require_relative '../lib/person_memberships'
+require_relative '../lib/cookie'
+
+describe PersonMemberships do
+  around { |test| VCR.use_cassette(File.basename(list_url), &test) }
+
+  let(:profile_url) { 'http://www.congreso.es/portal/page/portal/Congreso/Congreso/Diputados/BusqForm?_piref73_1333155_73_1333154_1333154.next_page=/wc/fichaDiputado?idDiputado=338&idLegislatura=12' }
+  let(:list_url) { 'http://www.congreso.es/portal/page/portal/Congreso/Congreso/Diputados/BusqForm?_piref73_1333155_73_1333154_1333154.next_page=/wc/listadoFichas?idDiputado=338&idLegislatura=12' }
+
+  # To get the correct response, the request has to be sent with the correct cookie.
+  # The cookie is set in the response from the membership url.
+  let(:profile_page_response) { open(profile_url) }
+  let(:cookie) { Cookie.new(profile_page_response.meta['set-cookie'])}
+  # We can now send a request to the memberships page with correct cookie.
+  let(:header){ { 'Cookie' => cookie.to_s } }
```

Comment:
I don't think you need the `Cookie` class at all. You can just pass the cookie back exactly as you received it:

```ruby
let(:header){ { 'Cookie' => profile_page_response.meta['set-cookie'] } }
```

However in my experimenting it seemed that there wasn't _always_ a cookie in the response, so we should probably only be sending this when there's a cookie in the response.

Here's the script I've been using to experiment with the cookies that are being returned:

```ruby
require 'open-uri'

def error?(res)
  if res.read.include?('missing string: error')
    warn "! Error"
  else
    warn "OK"
  end
end

memberships_url = 'http://www.congreso.es/portal/page/portal/Congreso/Congreso/Diputados/BusqForm?next_page=/wc/listadoFichas?idDiputado=338&idLegislatura=12'

warn "Requesting member page"
membership_page = open('http://www.congreso.es/portal/page/portal/Congreso/Congreso/Diputados/BusqForm?next_page=/wc/fichaDiputado?idDiputado=338&idLegislatura=12')

if membership_page.meta['set-cookie'].to_s.empty?
  warn "No Set-Cookie response header from membership page"
  warn "Requesting person memberships without Cookie"
  res = open(memberships_url)
  error?(res)
else
  warn "Requesting person memberships with Cookie"
  res = open(memberships_url, 'Cookie' => membership_page.meta['set-cookie'])
  error?(res)
end
```